File: reverse-linked-list/ReverseLinkedList.swift
class ReverseLinkedList {
    class LinkedList {
        var value: Int?
        var next: LinkedList?
    }

    func reverseLinkedList(head: LinkedList) -> LinkedList {
        var current = head
        var previous: LinkedList? = nil
        while(current != nil) {
            let next = current.next
            current.next = previous
            previous = current
            current = next
        }
        return previous
    }
}

File: reverse-linked-list/ReverseLinkedList.cs
public class ReverseLinkedList {
    static class LinkedList {
        public int value;
        public LinkedList next;
        public LinkedList(int value, LinkedList next) {
            this.value = value;
            this.next = next;
        }
    }

    public LinkedList ReverseLinkedLists(LinkedList head) {
        LinkedList current = head;
        LinkedList previous = null;
        while(current != null) {
            LinkedList next = current.next;
            current.next = previous;
            previous = current;
            current = next;
        }
        return previous;    
    }
}